components/bl-granim-component/src/index.js

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useMemo, useRef } from "react";
+
+import s from "./lib/granim.min";
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const COLORS_DEFAULT = [
+  ["#ff9966", "#ff5e62"],
+  ["#00F260", "#0575E6"],
+  ["#e1eec3", "#f05053"],
+];
+
+export default function GranimComponent({ component, eventHandlers, pods }) {
+  const {
+    style,
+    classList,
+    direction,
+    x0,
+    y0,
+    x1,
+    y1,
+    colors,
+    imageSource,
+    imagePosX,
+    imagePosY,
+    stretchModeX,
+    stretchModeY,
+    blendingMode,
+    isPausedWhenNotInView,
+    scrollDebounceThreshold,
+    transitionSpeed,
+    loop,
+  } = component;
+
+  const { onStart, onGradientChange, onEnd } = eventHandlers;
+
+  const granimContentPod = pods["granimContent"];
+
+  const canvasRef = useRef(null);
+  const canvasId = useMemo(() => BackendlessUI.UUID.short(), []);
+
+  const customDirection =
+    direction === "custom"
+      ? {
+          x0: x0 || "0px",
+          y0: y0 || "0px",
+          x1: x1 || "100%",
+          y1: y1 || "100%",
+        }
+      : {};
```

Comment:
create a hook useDirection which accept all props needed for calculate this, and move it to the bottom of component